lib/core/src/sdk.rs

```diff
@@ -286,22 +289,123 @@ impl LiquidSdk {
                             Ok(pending_chain_swaps) => {
                                 for swap in pending_chain_swaps {
                                     if let Err(e) = cloned.check_chain_swap_expiration(&swap).await {
-                                        error!("Error checking expiration for Send Swap {}: {e:?}", swap.id);
+                                        error!("Error checking expiration for Chain Swap {}: {e:?}", swap.id);
                                     }
                                 }
                             }
-                            Err(e) => error!("Error listing pending send swaps: {e:?}"),
+                            Err(e) => error!("Error listing pending chain swaps: {e:?}"),
                         }
                     },
                     _ = shutdown_receiver.changed() => {
-                        info!("Received shutdown signal, exiting refundable swaps loop");
+                        info!("Received shutdown signal, exiting pending swaps loop");
                         return;
                     }
                 }
             }
         });
     }
 
+    async fn track_chain_swap_transactions(self: &Arc<LiquidSdk>) {
+        let cloned = self.clone();
+        tokio::spawn(async move {
+            let mut shutdown_receiver = cloned.shutdown_receiver.clone();
+            let mut interval = tokio::time::interval(Duration::from_secs(60 * 10));
+            interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
+
+            loop {
+                tokio::select! {
+                    _ = interval.tick() => {
+                        if let Err(e) = cloned.check_chain_swap_refund_transactions().await {
+                            error!("Error checking chain swap refund transactions: {e:?}");
+                        }
+                        if let Err(e) = cloned.check_expired_chain_swap_lockup_transactions().await {
+                            error!("Error checking expired chain swap lockup tranactions: {e:?}");
+                        }
+                    },
+                    _ = shutdown_receiver.changed() => {
+                        info!("Received shutdown signal, exiting chain swap transactions loop");
+                        return;
+                    }
+                }
+            }
+        });
+    }
+
+    async fn check_chain_swap_refund_transactions(&self) -> Result<()> {
```

Comment:
I wonder whether we really need this function/tracking.
If the goal is to set the final state of the receive chain swap (Failed/Complete) then I have the following in my mind:
1. The Complete status is set after our claim (liquid) is confirmed and the place we do for lightning swaps is here: https://github.com/breez/breez-liquid-sdk/blob/main/lib/core/src/sdk.rs#L1193 perhaps it is also the place to verify the claim for chain swaps?
2. The Failed status indeed needs to be set when the refund tx is confirmed so perhaps we can wire it to the `check_expired_chain_swap_lockup_transactions` that already checks the expired swaps? maybe set to Failed if unspent_sat=0?

Reply:
There are two functions for the tracking in `check_chain_swap_refund_transactions()`:
1. In the case of a refund during the swap, mark the swap Failed once the lockup is spent. This can be cooperative refunds also, not just expired/non-cooperative. This is only for incoming chain swaps as the outgoing refund is [tracked here](https://github.com/breez/breez-liquid-sdk/blob/a9d75a06727599f813c4dd029150feb3ee03c966/lib/core/src/sdk.rs#L1325-L1331).
2. In the case extra funds sent to the lookup address (outside of the swap flow) and refunds are manually triggered (set back to Pending), mark successful/unsuccessful swap back to Complete/Failed once the funds are spent.

> 1. The Complete status is set after our claim (liquid) is confirmed and the place we do for lightning swaps is here: https://github.com/breez/breez-liquid-sdk/blob/main/lib/core/src/sdk.rs#L1193 perhaps it is also the place to verify the claim for chain swaps?

We could instead move the confirmation (set to Complete) of the incoming chain swap liquid claim here. We also don't monitor the confirmation of the outgoing chain swap bitcoin claim. Maybe we should?

> 2. The Failed status indeed needs to be set when the refund tx is confirmed so perhaps we can wire it to the `check_expired_chain_swap_lockup_transactions` that already checks the expired swaps? maybe set to Failed if unspent_sat=0?

We can but in this case it would only be the non-cooperative / expired refunds.

I main reason for the separation into two functions is to make sure refunds are always handled if the refund is pending.